test/test_lexer.ml

```diff
@@ -0,0 +1,50 @@
+open Ts_compiler
+open OUnit2
+
+let rec collect l acc =
+  match Lexer.next_token l with
```

Comment:
I had to actually implement a lexer unfortunately lol. I had to make sure this setup was actually working, and I got all green so we should be good to go from here